controllers/helmchart_controller_test.go

```diff
@@ -694,6 +694,40 @@ var _ = Describe("HelmChartReconciler", func() {
 					storage.ArtifactExist(*got.Status.Artifact)
 			}, timeout, interval).Should(BeTrue())
 
+			When("Setting valid valuesFiles attribute", func() {
+				updated := &sourcev1.HelmChart{}
+				Expect(k8sClient.Get(context.Background(), key, updated)).To(Succeed())
+				chart.Spec.ValuesFiles = []string{
+					"./charts/helmchart/values.yaml",
+					"./charts/helmchart/override.yaml",
+				}
+				Expect(k8sClient.Update(context.Background(), updated)).To(Succeed())
+				got := &sourcev1.HelmChart{}
+				Eventually(func() bool {
+					_ = k8sClient.Get(context.Background(), key, got)
+					return got.Status.Artifact != nil &&
+						storage.ArtifactExist(*got.Status.Artifact)
+				}, timeout, interval).Should(BeTrue())
+				helmChart, err := loader.Load(storage.LocalPath(*got.Status.Artifact))
```

Comment:
This artifact is never updated. I think these tests have been broken for a long time? This PR works in an actual deployment but the tests fail.

Reply:
Ran a test on `origin/main` printing our `updated`, `chart` and `got` (after resolution) and noticed that `got` does update with the overridden `ValuesFile`. I'm not sure if the test was ever intended to be able to be used this way though?

updated: 

```
&{
  {HelmChart source.toolkit.fluxcd.io/v1beta1} 
  {helmchart-sample-sxa8u  test-git-repository-fpb5b /apis/source.toolkit.fluxcd.io/v1beta1/namespaces/test-git-repository-fpb5b/helmcharts/helmchart-sample-sxa8u a5f66ee4-6f4a-4c05-8782-4edb9800c051 594 1 2021-04-04 16:18:18 -0400 EDT <nil> <nil> map[] map[] [] [finalizers.fluxcd.io]  []} 
  {helmchart-0.1.0.tgz * { GitRepository git-repository-sample-muy83} {3s}  false} {1 [{Ready True 0 2021-04-04 16:18:18 -0400 EDT ChartPackageSucceeded Fetched and packaged revision: 0.1.0}] http://localhost:5050/helmchart/test-git-repository-fpb5b/helmchart-sample-sxa8u/helmchart-latest.tgz 0xc000bc12c0 {}}
}
```

chart: 

```
&{
  { }
  {helmchart-sample-sxa8u  test-git-repository-fpb5b /apis/source.toolkit.fluxcd.io/v1beta1/namespaces/test-git-repository-fpb5b/helmcharts/helmchart-sample-sxa8u a5f66ee4-6f4a-4c05-8782-4edb9800c051 580 1 2021-04-04 16:18:18 -0400 EDT <nil> <nil> map[] map[] [] []  []} 
  {helmchart-0.1.0.tgz * { GitRepository git-repository-sample-muy83} {3s} ./charts/helmchart/override.yaml false} {0 []  <nil> {}}
}
```

got: 

```
&{
  {HelmChart source.toolkit.fluxcd.io/v1beta1} 
  {helmchart-sample-sxa8u  test-git-repository-fpb5b /apis/source.toolkit.fluxcd.io/v1beta1/namespaces/test-git-repository-fpb5b/helmcharts/helmchart-sample-sxa8u a5f66ee4-6f4a-4c05-8782-4edb9800c051 594 1 2021-04-04 16:18:18 -0400 EDT <nil> <nil> map[] map[] [] [finalizers.fluxcd.io]  []} 
  {helmchart-0.1.0.tgz * { GitRepository git-repository-sample-muy83} {3s}  false} {1 [{Ready True 0 2021-04-04 16:18:18 -0400 EDT ChartPackageSucceeded Fetched and packaged revision: 0.1.0}] http://localhost:5050/helmchart/test-git-repository-fpb5b/helmchart-sample-sxa8u/helmchart-latest.tgz 0xc000e83c80 {}}
}
```

Reply:
I've opened https://github.com/fluxcd/source-controller/pull/332 to nail down the tests.